spec/rubocop/cop/sorbet/t_struct_property_attr_macro_spec.rb

```diff
@@ -0,0 +1,297 @@
+# frozen_string_literal: true
+
+RSpec.describe(RuboCop::Cop::Sorbet::TStructPropertyAttrMacro, :config) do
+  context("when using `const` property") do
+    context("and `attr_reader` is used") do
+      context("and the names match") do
+        it("registers an offense") do
+          expect_offense(<<~RUBY)
+            class Foo < T::Struct
```

Comment:
We should detect this in other classes as well (`T::ImmutableStruct`, and probably others too).